WooCommerce/Classes/ViewRelated/Orders/Order Creation/PaymentSection/FeeOrDiscountLineDetailsViewModel.swift

```diff
@@ -30,44 +101,44 @@ class FeeLineDetailsViewModel: ObservableObject {
         }
     }
 
-    /// Decimal value of currently entered fee. For percentage type it is calculated final amount.
+    /// Decimal value of currently entered fee or discount. For percentage type it is calculated final amount.
     ///
     private var finalAmountDecimal: Decimal {
-        let inputString = feeType == .fixed ? amount : percentage
+        let inputString = feeOrDiscountType == .fixed ? amount : percentage
         guard let decimalInput = currencyFormatter.convertToDecimal(inputString) else {
             return .zero
         }
 
-        switch feeType {
+        switch feeOrDiscountType {
         case .fixed:
             return decimalInput as Decimal
         case .percentage:
             return baseAmountForPercentage * (decimalInput as Decimal) * 0.01
```

Comment:
Not for this PR, but would make sense for this line to be its own method in the CurrencyFormatter class? It doesn't feel the LineDetailsViewModel should know any details of how a percentage is calculated

Reply:
In my opinion, the view model is responsible for converting the UI input (percentage) and doing the calculation to provide the final discount amount. Especially since this is just one line and we save the overhead of a method in another class. On top of that, the operation is not duplicated elsewhere.